addons/point_of_sale/static/src/app/models/product_product.js

```diff
@@ -0,0 +1,224 @@
+/** @odoo-module */
+import { registry } from "@web/core/registry";
+import { Base } from "./related_models";
+import { deserializeDate } from "@web/core/l10n/dates";
+import { _t } from "@web/core/l10n/translation";
+import { roundPrecision } from "@web/core/utils/numbers";
+
+const { DateTime } = luxon;
+/**
+ * ProductProduct, shadow of product.product in python.
+ * To works properly, this model needs to be registered in the registry
+ * with the key "pos_available_models". And to be instanciated with the
+ * method createRelatedModels from related_models.js
+ *
+ * Models to load: product.product, uom.uom
+ */
+
+export class ProductProduct extends Base {
+    static pythonModel = "product.product";
+
+    // Getter
+    isAllowOnlyOneLot() {
+        const productUnit = this.uom_id;
+        return this.tracking === "lot" || !productUnit || !productUnit.is_pos_groupable;
+    }
+
+    isTracked() {
+        const stock_picking_type = this.models["stock.picking.type"].readAll()[0];
+
+        return (
+            ["serial", "lot"].includes(this.tracking) &&
+            (stock_picking_type.use_create_lots || stock_picking_type.use_existing_lots)
+        );
+    }
+
+    async _onScaleNotAvailable() {}
+
+    isConfigurable() {
+        const attrByIds = this.models["product.attribute"].getAllBy("id");
+        return this.attribute_line_ids.some((line) => line.attribute_id?.id in attrByIds);
+    }
+
+    get isScaleAvailable() {
+        return true;
+    }
+
+    get parentCategoryIds() {
```

Comment:
in JS, if we want to return the record objects, do not put `ids` suffix. E.g. the name of the getter should be `parentCategories`.